app/controllers/users_controller.rb

```diff
@@ -1900,6 +1900,11 @@ def user_menu_messages
            }
   end
 
+  def lookup
+    users = User.where(username: params[:usernames])
+    render json: serialize_found_users(users)
+  end
+
```

Comment:
```
import userSearch from "discourse/lib/user-search";
```


Was not working out for you?

Reply:
It wasn't, `userSearch` takes a search term (a part of a username) and returns found users. While in this PR I have a list of usernames, for example ["user1", "user2", "moderator", "admin"] and I need to look up and load those users.